City-of-Helsinki/berth-reservations
# VEN-546 | Admin Timmi data importer

## Description :sparkles:
* Add an admin importer for the JSON generated by the Timmi parser

## Issues :bug:
### Closes :no_good_woman:
**[VEN-546](https://helsinkisolutionoffice.atlassian.net/browse/VEN-546):** add data importer to Django admin

### Related :handshake:
* #213
* https://github.com/City-of-Helsinki/open-city-profile/pull/118

## Testing :alembic:
### Automated tests :gear:️
```shell
$ pytest customers/tests/test_customer_models.py::test_import_customer_data_with_valid_data_set
$ pytest customers/tests/test_customer_models.py::test_import_customer_data_missing_id
```

### Manual testing :construction_worker_man:
1. Take the JSON from the `test_import_customer_data_with_valid_data_set` and save it as `customers.json`
2. Go to http://localhost:8000/admin/customers/customerprofile/upload-json/
3. Select the `customers.json` file and hit upload
4. It should return a `export.json` file with the ids of the imported customers
```json
{"313431": "98edea83-4c92-4dda-bb90-f22e9dafe94c", "313432": "48319ebc-5eaf-4285-a565-15848225614b"}
```

## Screenshots :camera_flash:
<img width="866" alt="image" src="https://user-images.githubusercontent.com/15201480/91559851-7ac37f80-e941-11ea-8637-81675bd2dc04.png">
